src/components/PublisherView.js

```diff
@@ -16,40 +16,49 @@ const publishListeners = ['onPublishStart', 'onPublishStop', 'onPublishError'];
 const NOOP = () => {};
 
 export default class PublisherView extends React.Component<
-  PublisherViewPropsWithListeners
+  PublisherViewProps,
+  { camera: number }
 > {
-  props: PublisherViewPropsWithListeners;
-
   static defaultProps = {
     onPublishStart: NOOP,
     onPublishStop: NOOP,
     onPublishError: NOOP,
     sessionId: '',
+    audioDisabled: false,
+    videoDisabled: false,
+  };
+
+  state = {
+    camera: 0,
   };
 
   componentWillMount() {
-    publishListeners.forEach(listener => this.addListener(listener));
+    publishListeners.forEach(listener => this._addListener(listener));
   }
 
   componentWillUnmount() {
-    publishListeners.forEach(listener => this.removeListener(listener));
+    publishListeners.forEach(listener => this._removeListener(listener));
   }
 
-  addListener = (name: string) => {
+  _addListener = (name: string) => {
```

Comment:
There is no need for underscoring the methods because is not the public component.